server/src/test/java/io/druid/server/coordinator/LoadQueuePeonTest.java

```diff
@@ -0,0 +1,355 @@
+      Assert.assertTrue(timing.forWaiting().awaitLatch(dropRequestSignal[requestSignalIdx.get()]));
+      Assert.assertNotNull(curator.checkExists().forPath(dropRequestPath));
+      Assert.assertEquals(
+          segment,
+          ((SegmentChangeRequestDrop) jsonMapper.readValue(
+              curator.getData()
+                     .decompressed()
+                     .forPath(dropRequestPath), DataSegmentChangeRequest.class
+          )).getSegment()
+      );
+
+      if (requestSignalIdx.get() == 4) {
+        requestSignalIdx.set(0);
+      } else {
+        requestSignalIdx.incrementAndGet();
+      }
+
+      // simulate completion of drop request by historical
+      curator.delete().guaranteed().forPath(dropRequestPath);
+      Assert.assertTrue(timing.forWaiting().awaitLatch(segmentDroppedSignal[segmentSignalIdx.get()]));
+
+      int expectedNumSegmentToDrop = 5 - segmentSignalIdx.get() - 1;
+      Assert.assertEquals(expectedNumSegmentToDrop, loadQueuePeon.getSegmentsToDrop().size());
+
+      if (segmentSignalIdx.get() == 4) {
+        segmentSignalIdx.set(0);
+      } else {
+        segmentSignalIdx.incrementAndGet();
+      }
+    }
+
+    for (DataSegment segment : segmentToLoad) {
+      String loadRequestPath = ZKPaths.makePath(LOAD_QUEUE_PATH, segment.getIdentifier());
+      Assert.assertTrue(timing.forWaiting().awaitLatch(loadRequestSignal[requestSignalIdx.get()]));
+      Assert.assertNotNull(curator.checkExists().forPath(loadRequestPath));
+      Assert.assertEquals(
+          segment, ((SegmentChangeRequestLoad) jsonMapper.readValue(
+              curator.getData()
+                     .decompressed()
+                     .forPath(loadRequestPath),
+              DataSegmentChangeRequest.class
+          )).getSegment()
+      );
+
+      requestSignalIdx.incrementAndGet();
+
+      // simulate completion of load request by historical
+      curator.delete().guaranteed().forPath(loadRequestPath);
+      Assert.assertTrue(timing.forWaiting().awaitLatch(segmentLoadedSignal[segmentSignalIdx.get()]));
+
+      int expectedNumSegmentToLoad = 5 - segmentSignalIdx.get() - 1;
+      Assert.assertEquals(1200 * expectedNumSegmentToLoad, loadQueuePeon.getLoadQueueSize());
+      Assert.assertEquals(expectedNumSegmentToLoad, loadQueuePeon.getSegmentsToLoad().size());
+      segmentSignalIdx.incrementAndGet();
+    }
+  }
+
+  @Test
+  public void testFailAssign() throws Exception
+  {
+    final DataSegment segment = dataSegmentWithInterval("2014-10-22T00:00:00Z/P1D");
+
+    final CountDownLatch loadRequestSignal = new CountDownLatch(1);
+    final CountDownLatch segmentLoadedSignal = new CountDownLatch(1);
+
+    loadQueuePeon = new LoadQueuePeon(
+        curator,
+        LOAD_QUEUE_PATH,
+        jsonMapper,
+        Execs.scheduledSingleThreaded("test_load_queue_peon_scheduled-%d"),
+        Execs.singleThreaded("test_load_queue_peon-%d"),
+        new TestDruidCoordinatorConfig()
```

Comment:
see other comment regarding making TestDruidCoordinatorConfig(..) more generic so that you don't have to create anonymous class here.